dotcom-rendering/src/web/components/SignInGate/gateDesigns/SignInGateMainCheckoutComplete.tsx

```diff
@@ -0,0 +1,199 @@
+const COMPLETE_REGISTRATION_BODY_NO_ADS_INCENTIVE =
+	'Complete your registration to receive fewer requests for financial support, to easily manage your account, and to subscribe to newsletters and comment. ';
+const SIGN_IN_BODY_ADS_INCENTIVE =
+	'Sign in to stop seeing ads, to see fewer requests for financial support, to subscribe to newsletters and comment, and to easily manage your account. ';
+const SIGN_IN_BODY_NO_ADS_INCENTIVE =
+	'Sign in to receive fewer requests for financial support, to easily manage your account, and to subscribe to newsletters and comment. ';
+
+// BUTTON TEXT
+const COMPLETE_REGISTRATION_BUTTON = 'Complete registration';
+const SIGN_IN_BUTTON = 'Sign in';
+
+const getHeadingText: (userType: UserType) => string = (userType) => {
+	const headingMap: Record<UserType, string> = {
+		new: COMPLETE_REGISTRATION_HEADER,
+		guest: COMPLETE_REGISTRATION_HEADER,
+		current: SIGN_IN_HEADER,
+	};
+	return headingMap[userType];
+};
+
+const getSubHeadingText: (product: Product) => string = (product) => {
+	const subHeadingMap: Record<Product, string> = {
+		DigitalPack: SUBSCRIPTION_SUBHEADER,
+		Paper: SUBSCRIPTION_SUBHEADER,
+		GuardianWeekly: SUBSCRIPTION_SUBHEADER,
+		Contribution: SUPPORTER_SUBHEADER,
+	};
+	return subHeadingMap[product];
+};
+
+const getBodyText: (
+	checkoutCompleteCookieData: CheckoutCompleteCookieData,
+) => string = (checkoutCompleteCookieData) => {
+	const newOrGuestUserBodyMap: Record<Product, string> = {
+		DigitalPack: COMPLETE_REGISTRATION_BODY_ADS_INCENTIVE,
+		Paper: COMPLETE_REGISTRATION_BODY_NO_ADS_INCENTIVE,
+		Contribution: COMPLETE_REGISTRATION_BODY_NO_ADS_INCENTIVE,
+		GuardianWeekly: COMPLETE_REGISTRATION_BODY_NO_ADS_INCENTIVE,
+	};
+
+	const currentUserBodyMap: Record<Product, string> = {
+		DigitalPack: SIGN_IN_BODY_ADS_INCENTIVE,
+		Paper: SIGN_IN_BODY_NO_ADS_INCENTIVE,
+		Contribution: SIGN_IN_BODY_NO_ADS_INCENTIVE,
+		GuardianWeekly: SIGN_IN_BODY_NO_ADS_INCENTIVE,
+	};
+
+	const { userType, product } = checkoutCompleteCookieData;
+
+	switch (userType) {
+		case 'new':
+			return newOrGuestUserBodyMap[product];
+		case 'guest':
+			return newOrGuestUserBodyMap[product];
+		case 'current':
+			return currentUserBodyMap[product];
+	}
+};
+
+const getButtonText: (userType: UserType) => string = (userType) => {
+	const buttonMap: Record<UserType, string> = {
+		new: COMPLETE_REGISTRATION_BUTTON,
+		guest: COMPLETE_REGISTRATION_BUTTON,
+		current: SIGN_IN_BUTTON,
+	};
+	return buttonMap[userType];
+};
+
+export const SignInGateMainCheckoutComplete = ({
+	signInUrl,
+	guUrl,
+	dismissGate,
+	abTest,
+	ophanComponentId,
+	isMandatory = false,
+	checkoutCompleteCookieData,
+}: SignInGatePropsWithCheckoutCompleteCookieData) => {
+	const { userType, product } = checkoutCompleteCookieData;
+
+	// send new/guest userType to the /register page instead of /signin
+	const personaliseSignInURl = (url: string): string => {
```

Comment:
Small point, but I would maybe more explicit about what the function does. Eg, changeSignInLinkToRegister or something. 

I would also perhaps do an explicit check that the user is 'guest' or 'current' before doing the swap and making the other case default. Very minor point though, as we should be safe here with the type checking that happens further up.

Reply:
Changed my mind about the naming of the method, but did switch over the logic here